packages/firebase_vertexai/firebase_vertexai/lib/src/vertex_model.dart

```diff
@@ -232,9 +230,37 @@ final class GenerativeModel {
   Future<BatchEmbedContentsResponse> batchEmbedContents(
       Iterable<EmbedContentRequest> requests) async {
     return _googleAIModel
-        .batchEmbedContents(
-            requests.map((e) => e._toGoogleAIEmbedContentRequest()))
-        .then(
-            BatchEmbedContentsResponse._fromGoogleAIBatchEmbedContentsResponse);
+        .batchEmbedContents(requests.map((e) => e.toGoogleAI()))
+        .then((r) => r.toVertex());
   }
 }
+
+/// Conversion utilities for [GenerativeModel].
+extension GoogleAIGenerativeModelConversion on GenerativeModel {
+  /// Return this model as a [google_ai.GenerativeModel].
+  google_ai.GenerativeModel get googleAIModel => _googleAIModel;
+}
+
+/// Returns a [GenerativeModel] using it's private constructor.
+GenerativeModel createGenerativeModel({
+  required FirebaseApp app,
+  required String location,
+  required String model,
+  Content? systemInstruction,
+  FirebaseAppCheck? appCheck,
```

Comment:
I had confused myself during the merge and added the argument in the wrong place, but I think it's cleaned up now. This should forward directly because it's taking place of the private constructor in uasge.